tests/unit/domain/getZip.test.ts

```diff
@@ -0,0 +1,160 @@
+    const response = await getZip({} as ZipDetails, ({} as unknown) as S3, 'bucket', undefined, 'test');
+    const error = new MissingFolderNameError();
+
+    expect(response.statusCode).toBe(500);
+    expect(response.body).toEqual(error.message);
+  });
+
+  it('returns a bad request if the file name is invalid', async () => {
+    const event: ZipDetails = {
+      adrDocumentId: undefined,
+    };
+    const response = await getZip(event, ({} as unknown) as S3, 'bucket', 'folder', 'test');
+    const error = new FileNameError();
+
+    expect(response.statusCode).toBe(400);
+    expect(response.body).toEqual(error.message);
+  });
+
+  it('returns an internal server error if there is no Body in the S3 request', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const mockPromise = jest.fn().mockReturnValue(Promise.resolve({ ContentType: 'application/octet-stream' }));
+    const mockGetObject = jest.fn().mockReturnValue({ promise: mockPromise });
+
+    mockS3.getObject = mockGetObject;
+
+    const event: ZipDetails = {
+      adrDocumentId: '1234',
+    };
+    const response = await getZip(event, mockS3, 'bucket', 'folder', 'test');
+    const error = new NoBodyError();
+
+    expect(response.statusCode).toBe(500);
+    expect(response.body).toEqual(error.message);
+  });
+
+  it('returns an 404 if the stored file is not a ZIP', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const mockPromise = jest
+      .fn()
+      .mockReturnValue(Promise.resolve({ Body: 'This is an image', ContentType: 'image/jpg' }));
+    const mockGetObject = jest.fn().mockReturnValue({ promise: mockPromise });
+
+    mockS3.getObject = mockGetObject;
+
+    const event: ZipDetails = {
+      adrDocumentId: '1234',
+    };
+    const response = await getZip(event, mockS3, 'bucket', 'folder', 'test');
+    const error = new IncorrectFileTypeError();
+
+    expect(response.statusCode).toBe(404);
+    expect(response.body).toEqual(error.message);
+  });
+
+  it('returns a not found error if the file is not found', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const mockPromise = jest.fn().mockReturnValue(Promise.reject(({ code: 'NoSuchKey' } as unknown) as Error)); // eslint-disable-line prefer-promise-reject-errors
+    const mockGetObject = jest.fn().mockReturnValue({ promise: mockPromise });
+
+    mockS3.getObject = mockGetObject;
+
+    const event: ZipDetails = {
+      adrDocumentId: '1234',
+    };
+    const response = await getZip(event, mockS3, 'bucket', 'folder', 'test');
+
+    expect(response.statusCode).toBe(404);
+    expect(response.body).toBe('NoSuchKey');
+  });
+
+  it('returns an internal server error if the S3 get fails for any other reason', async () => {
+    const mockS3 = ({} as unknown) as S3;
+    const mockPromise = jest.fn().mockReturnValue(Promise.reject(({ code: 'Generic Error' } as unknown) as Error)); // eslint-disable-line prefer-promise-reject-errors
```

Comment:
aren't you already doing promise reject, why is it erroring?

Reply:
Not sure I follow what you are asking here? As with the other route tests, this test is to cover the catch block instances where the error is not a NoBodyError / MissingBucketNameError / MissingFolderNameError / FileNameError / IncorrectFileTypeError.

Reply:
on about the eslint line that's being switched off

Reply:
Isn't that in the getZip test below?

Reply:
I see what you mean, removed.